File: tmp_api_probe.py
import urllib.request
import urllib.error
import json

urls = ['https://DevNumb-MLYorkchillerOptimzer.hf.space', 'https://huggingface.co/spaces/DevNumb/MLYorkchillerOptimzer']
for url in urls:
    print('URL:', url)
    for path in ['/health', '/', '/predict']:
        full = url.rstrip('/') + path
        try:
            req = urllib.request.Request(full, method='GET')
            with urllib.request.urlopen(req, timeout=10) as r:
                body = r.read(5000).decode('utf-8', errors='replace')
                print('  GET', path, 'OK', r.status, body[:200].replace('\n', ' '))
        except Exception as e:
            print('  GET', path, 'ERR', repr(e))
    print()

sample = {'features': [1200, 18.6, 6.5, 90, 10, 5, 0, 1, 1, 0, 0, 2]}
formats = [
    ('array', sample),
    ('named', {
        'cooling_load': 1200,
        'wet_bulb_temp': 18.6,
        'chw_setpoint': 6.5,
        'current_limit': 90,
        'hour': 10,
        'month': 5,
        'weekend': 0,
        'chiller1': 1,
        'chiller2': 1,
        'chiller3': 0,
        'chiller4': 0,
        'total_chillers': 2,
    }),
    ('data', {'data': [[1200, 18.6, 6.5, 90, 10, 5, 0, 1, 1, 0, 0, 2]]}),
]
for fmt, payload in formats:
    print('POST format', fmt)
    try:
        req = urllib.request.Request(
            'https://DevNumb-MLYorkchillerOptimzer.hf.space/predict',
            data=json.dumps(payload).encode('utf-8'),
            headers={'Content-Type': 'application/json'},
            method='POST',
        )
        with urllib.request.urlopen(req, timeout=15) as r:
            body = r.read(1000).decode('utf-8', errors='replace')
            print('  OK', r.status, body[:400].replace('\n', ' '))
    except urllib.error.HTTPError as e:
        print('  HTTPERR', e.code, e.read(1000).decode('utf-8', errors='replace')[:400].replace('\n', ' '))
    except Exception as e:
        print('  ERR', repr(e))

File: tmp_api_probe.js

const API_BASE_URL = 'https://DevNumb-randomforestmodel.hf.space/gradio_api/call/predict';

async function testAPI(features, label) {
  console.log(`Testing ${label}...`);
  try {
    const postResponse = await fetch(API_BASE_URL, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ data: features })
    });
    const postData = await postResponse.json();
    const { event_id } = postData;
    await new Promise(r => setTimeout(r, 2000));
    const getResponse = await fetch(`${API_BASE_URL}/${event_id}`);
    const text = await getResponse.text();
    const dataMatch = text.match(/data:\s*\["([^"]+)"\]/);
    console.log(`${label} Result:`, dataMatch ? dataMatch[1] : 'No match');
  } catch (e) {
    console.error(`${label} failed:`, e.message);
  }
}

async function runTests() {
  await testAPI([70, 55, 14, 3, 7, 1, 1, 1, 80, 80, 80, 60, 60, 60, 280, 280, 280, 1200], 'Scenario 70F/1200T');
  await testAPI([95, 75, 14, 3, 7, 1, 1, 1, 80, 80, 80, 60, 60, 60, 280, 280, 280, 1200], 'Scenario 95F/1200T');
  await testAPI([85, 65, 14, 3, 7, 1, 1, 1, 80, 80, 80, 60, 60, 60, 280, 280, 280, 500], 'Scenario 85F/500T');
  await testAPI([85, 65, 14, 3, 7, 1, 1, 1, 80, 80, 80, 60, 60, 60, 280, 280, 280, 2000], 'Scenario 85F/2000T');
}

runTests();
